src/components/PopupWithDeleteConfirm.js

```diff
@@ -0,0 +1,24 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithDeleteConfirm extends Popup {
+  constructor(popupSelector, handleDeleteSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._handleCardDelete = handleDeleteSubmit;
+  }
+
+  open(cardId, cardElement) {
+    super.open();
+    this._cardId = cardId;
```

Comment:
Needs correction: `PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```